GeosCore/tracer_mod.F90

```diff
@@ -0,0 +1,760 @@
+       SpcInfo => State_Chm%SpcData(N)%Info
+
+       ! Skip Rn-Pb-Be tracers for now
+       IF ( SpcInfo%Is_RadioNuclide ) CYCLE
+
+       ! Skip if no sink
+       IF ( TRIM(SpcInfo%Snk_Mode) == 'none' ) CYCLE
+
+       !---------------------------------------------------------------------
+       ! Compute the decay rate (if needed)
+       !---------------------------------------------------------------------
+       IF ( TRIM(SpcInfo%Snk_Mode) == 'efolding' .or. &
+            TRIM(SpcInfo%Snk_Mode) == 'halflife' ) THEN
+
+          IF ( TRIM(SpcInfo%Snk_Mode) == 'efolding' ) THEN
+
+             ! Compute the decay constant (s-1) from the e-folding time
+             !  ln(N/No) = ln(1/e) = (-1) =  -decayConstant * e-folding time
+             DecayConstant = 1.0 / ( SpcInfo%Snk_Period * Day2Sec )
+
+          ELSE IF ( TRIM(SpcInfo%Snk_Mode) == 'halflife' ) THEN
+
+             ! Compute the decay constant (s-1) from the half-life:
+             !  ln(N/No) = ln(1/2) = -decayConstant * halfLife
+             DecayConstant = ln2 / ( SpcInfo%Snk_Period * Day2Sec )
+
+          ENDIF
+
+          DecayRate  = EXP( - DecayConstant * DT )
+
+          !### Debug output
+          IF ( Input_Opt%Verbose ) THEN
+             WRITE( 6,100 ) ADJUSTL( SpcInfo%Name ), DecayRate
+ 100         FORMAT( '     -  Species name, decay rate: ', a15, es13.6 )
+          ENDIF
+
+       ENDIF
+
+       !--------------------------------------------------------------------
+       ! Get chemical loss (if needed)
+       !--------------------------------------------------------------------
+       IF ( TRIM(SpcInfo%Snk_Mode) == 'chemical_loss' ) THEN
+
+          ! For stOX tracer get O3 loss [molec/cm3/s] from HEMCO
+          IF ( TRIM(SpcInfo%Name ) == 'stOX' ) THEN
+
+             CALL HCO_GetPtr( HcoState, 'O3_LOSS', LOSS_O3,   RC )
+             IF ( RC /= GC_SUCCESS ) THEN
+                ErrMsg = 'Cannot get pointer to O3_LOSS!'
+                CALL GC_Error( ErrMsg, RC, ThisLoc )
+                RETURN
+             ENDIF
+
+          ELSE
+
+             ErrMsg = 'Snk_Mode: chemical_loss not currently supported '  // &
+                   'for ' // TRIM( SpcInfo%Name )  // &
+                   'Please modify species_database.yml or add this '      // &
+                   'capability in tracer_mod.F90.'
+             CALL GC_Error( ErrMsg, RC, ThisLoc )
+             RETURN
+
+          ENDIF
+
+       ENDIF
+
+       !---------------------------------------------------------------------
+       ! Determine mask
+       !---------------------------------------------------------------------
+       IF ( TRIM(SpcInfo%Snk_Horiz) == 'lat_zone' ) THEN
+
+          !$OMP PARALLEL DO                                                  &
+          !$OMP DEFAULT( SHARED                                             )&
+          !$OMP PRIVATE( I, J                                               )
+          DO J = 1, State_Grid%NY
```

Comment:
TODO FOR LATER: add `!$OMP COLLAPSE( 2 )` to vectorize the loop

Reply:
Now done in commit cdd5cd3d.